lib/segment/src/index/field_index/geo_hash.rs

```diff
@@ -73,11 +74,11 @@ fn sphere_neighbor(hash_str: &str, direction: Direction) -> Result<GeoHash, Geoh
     let lat = sphere_lat(coord.y + 2f64 * lat_err.abs() * dlat);
 
     let neighbor_coord = Coord { x: lon, y: lat };
-    encode(neighbor_coord, hash_str.len())
+    encode(neighbor_coord, hash_str.len()).map(SmolStr::from)
```

Comment:
Nitpick. It is convention to use `Into::into` here. We can do this because the type is inferred. This way we hide implementation details.

```suggestion
    encode(neighbor_coord, hash_str.len()).map(Into::into)
```

Reply:
I like this option, I don't see it as nitpick, I want to avoid `SmolStr` naming too